src/main/java/me/ryanhamshire/GriefPrevention/PlayerEventHandler.java

```diff
@@ -2620,6 +2636,21 @@ else if(playerData.shovelMode == ShovelMode.Subdivide)
 			}
 		}
 	}
+
+	// Stops an untrusted player from removing a book from a lectern
+	@EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
+	void onTakeBook(PlayerTakeLecternBookEvent event)
+	{
+		Player player = event.getPlayer();
+		PlayerData playerData = this.dataStore.getPlayerData(player.getUniqueId());
+		Claim claim = this.dataStore.getClaimAt(event.getLectern().getLocation(), false, playerData.lastClaim);
+		if (claim.allowContainers(player) != null)
```

Comment:
You need to nullcheck here.

Reply:
And if it isn't null, then you can 
- set player#lastClaim
- check allowContainers

Reply:
Oh duh, how did I forget that. I guess I didn't test outside of a claim to see what I would assume would be a nasty error.

Ill push my fixes in a minute after I test to make sure they work.